models/classes/counter/CounterService.php

```diff
@@ -0,0 +1,195 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types = 1);
+
+namespace oat\tao\model\counter;
+
+use common_Exception;
+use common_persistence_KeyValuePersistence as KeyValuePersistence;
+use oat\oatbox\event\Event;
+use oat\oatbox\event\EventManager;
+use oat\oatbox\service\ConfigurableService;
+use oat\oatbox\service\exception\InvalidServiceManagerException;
+use oat\oatbox\service\ServiceNotFoundException;
+
+class CounterService extends ConfigurableService
+{
+    public const SERVICE_ID = 'tao/CounterService';
+
+    public const OPTION_PERSISTENCE = 'persistence';
+    public const OPTION_COUNTER_KEY_PREFIX = 'counterKeyPrefix';
+    public const OPTION_EVENTS = 'events';
+
+    public const DEFAULT_PREFIX = 'tao:counter:';
+
+    protected const OPTION_CALLBACK = 'keyCallbackMethod';
+    protected const OPTION_SHORT_NAME = 'shortName';
```

Comment:
https://github.com/oat-sa/tao-core/pull/3101/files#r704079172

```suggestion
    private const OPTION_CALLBACK = 'keyCallbackMethod';
    private const OPTION_SHORT_NAME = 'shortName';
```